File: os2skoledata-ad-sync/Services/OS2skoledata/Model/ModificationHistory.cs
using System;
using System.Collections.Generic;

namespace os2skoledata_ad_sync.Services.OS2skoledata.Model
{
    public class ModificationHistory
    {
        public long Id { get; set; }
        public DateTime Tts { get; set; }
        public long EntityId { get; set; }
        public EntityType EntityType { get; set; }
        public EventType EventType { get; set; }
        public string EntityName { get; set; }
        public long InstitutionId { get; set; }
        public string InstitutionName { get; set; }

        // only for EntityType person
        public List<string> Groups { get; set; }
    }

    public enum EntityType
    {
        INSTITUTION,
        INSTITUTION_PERSON,
        GROUP
    }

    public enum EventType
    {
        CREATE,
        UPDATE,
        DELETE
    }
}

File: os2skoledata-azure-ad-sync/src/main/java/dk/digitalidentity/service/model/ModificationHistory.java
package dk.digitalidentity.service.model;

import dk.digitalidentity.service.model.enums.EntityType;
import dk.digitalidentity.service.model.enums.EventType;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
public class ModificationHistory {
	private Long id;
	private long entityId;
	private EntityType entityType;
	private EventType eventType;
	private String entityName;
	private long institutionId;
	private String institutionName;

	// only for EntityType person
	private List<String> groups;

}
